lib/floki.ex

```diff
@@ -95,6 +95,7 @@ defmodule Floki do
   defp raw_html([], html), do: html
   defp raw_html(tuple, html) when is_tuple(tuple), do: raw_html([tuple], html)
   defp raw_html([value|tail], html) when is_binary(value), do: raw_html(tail, html <> value)
+  defp raw_html([{elem, value}|tail], html) when elem == :comment, do: raw_html(tail, html <> "<!--#{value}-->")
```

Comment:
@glaucocustodio here you don't need to use a guard. Instead, you can pattern match the `elem`:

```elixir
defp raw_html([{:comment, value}|tail], html), do: raw_html(tail, html <> "<!--#{value}-->")
```